### DISCUSSION TITLE
Mesh Parameterization for specified surfaces

### DISCUSSION THREAD

**Discussion by AbhijithMoni** [mdolab/MACH-Aero#146] [Q&A] [answered]
The parameterization of the mesh for specific surfaces within a geometry fails to reflect the deformation applied by the FFD box. When using the `writeGrid` option, the resulting mesh tends to remain unchanged from its initial state, lacking the expected parameterization.

```

from mpi4py import MPI
from idwarp import USMesh
from pygeo import DVGeometry
import os

output_dir = "./"
os.makedirs(output_dir, exist_ok=True)

FFDFile = "M_FFD_opt.xyz"
gridFile = "uCRM_coarse.cgns"

meshOptions = {
    "gridFile": gridFile,
    "fileType": "CGNS",
    'specifiedSurfaces': [b'wingle', b'wingu', b'wingl', b'wingte'],
    "symmetrySurfaces": [b'symm'],
    'alpha': 0.3,
    'aExp': 2.0,
}

mesh = USMesh(options=meshOptions, comm=MPI.COMM_WORLD)
DVGeo = DVGeometry(FFDFile)

coords0 = mesh.getSurfaceCoordinates()

DVGeo.addPointSet(coords0, "coords")
dvDict = DVGeo.getValues()
DVGeo.setDesignVars(dvDict)

mesh.setSurfaceCoordinates(coords0)

mesh.warpMesh()

mesh.writeGrid('warped.cgns')
```

Any help is appreciated. Please review the code and suggest any necessary changes and files are attached. 

[Files.zip](https://github.com/mdolab/MACH-Aero/files/14669248/Files.zip)

Many thanks.

**Comment by sseraj** [comment]
You are setting the surface coordinates to `coords0`, which are the baseline coordinates. You need to get the updated points from `DVGeo.update` and set those as the surface coordinates.

**Reply by AbhijithMoni** [reply]
@sseraj Thank you for your input.  I attempted the suggested change, but it appears that there is no improvement in the result. I suspect there might be an issue with my implementation. Could you please provide any further guidance on adjustments needed in the code? 

```
mesh = USMesh(options=meshOptions, comm=MPI.COMM_WORLD)
DVGeo = DVGeometry(FFDFile)

coords0 = mesh.getSurfaceCoordinates()

DVGeo.addPointSet(coords0, "coords")
dvDict = DVGeo.getValues()
DVGeo.setDesignVars(dvDict)

mesh.setSurfaceCoordinates(DVGeo.update("coords"))

mesh.warpMesh()

mesh.writeGrid('warped.cgns')
```
Many thanks.

**Reply by sseraj** [reply]
It looks like you are not adding or changing any DVs here:
```
dvDict = DVGeo.getValues()
DVGeo.setDesignVars(dvDict)
```

**Reply by AbhijithMoni** [reply]
@sseraj My FFD box (`FFDFile = "M_FFD_opt.xyz"`) is already parameterized, does this imply that I should directly pass the DV positions to `DVGeo.setDesignVars` to parameterize the mesh?

**Reply by sseraj** [reply, marked as answer]
Yes, you should set the DVs on the baseline FFD. Embedding the surface in an optimized FFD will not deform the surface.

**Reply by AbhijithMoni** [reply]
@sseraj Thanks for clarifying.
